### PR TITLE
firestore: marshal values as []byte, not string

### LINKED ISSUE #4145: Firestore: Local User Setup Error
### Description

**What happened**:
1. On a fresh Teleport cluster, create a new local user. `tctl users add <user> --roles admin`
2. Visit user setup link. `https://<cluster_host>:3080/web/invite/d92a13c821739fe26ae4c3c501ce78f3`. Expired link... 
image
3. In Teleport debug logs, error fetching reset password token.
```
DEBU             pushing event Put for key '/web/users/<user>/params'. firestore/firestorebk.go:609
DEBU [USAGE]     Ignoring event "user.create". pro/auditlog.go:80
INFO [AUTH]      "<user>" user created auth/grpcserver.go:396
DEBU [USAGE]     Ignoring event "reset_password_token.create". pro/auditlog.go:80
DEBU             pushing event Put for key '/web/users/<user>/pwd'. firestore/firestorebk.go:609
DEBU             pushing event Put for key '/resetpasswordtokens/d92a13c821739fe26ae4c3c501ce78f3/params'. firestore/firestorebk.go:609
WARN             Failed to fetch a reset password token: rpc error: code = Internal desc = grpc: error while marshaling: proto: field "google.firestore.v1.Value.ValueType" contains invalid UTF-8. web/apiserver.go:1218
```

**What you expected to happen**:
User credential set up screen is presented. 

**How to reproduce it (as minimally and precisely as possible)**:
See above.

### Environment
- Teleport version (use `teleport version`): Teleport Enterprise v4.3.2git:v4.3.2-0-gc6d702ad8 go1.13.2
- Tsh version (use `tsh version`): N/A
- OS (e.g. from `/etc/os-release`): Ubuntu 16.04.6 LTS

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):
GCP

**Browser environment**

- Browser Version (for UI-related issues): Chrome 84.0.4147.89
- Install tools: N/A
- Others: N/A

**Relevant Debug Logs If Applicable**

- `tsh --debug`
- `teleport --debug`

### PR DESCRIPTION
**Pull request by awly**
Firestore marshaler requires all string data to be valid UTF-8. Our
values are sometimes binary (like QR codes for OTP signup), which would
fail to marhal.

On reads, try using both new and old formats to support existing data.

Added tests for this fallback and for writing binary data into all
backends.

Tested with firestore, including upgrade from an existing teleport cluster.

Fixes #4145

### CODE REVIEW

**Comment by webvictim** [comment]
Looks like the linter is failing.

**Comment by awly** [comment]
Fixed linter failures

File: lib/backend/firestore/firestorebk.go [hunk 1]
```diff
@@ -110,13 +110,63 @@ type FirestoreBackend struct {
 }
 
 type record struct {
+	Key       string `firestore:"key,omitempty"`
```
Comment (fspmarshall) [inline lib/backend/firestore/firestorebk.go:113]: Should `Key` be changed to `[]byte` as well while we're here?  I don't think we enforce UTF-8 on keys either.

**Review by fspmarshall — APPROVED** [review]

File: lib/backend/firestore/firestorebk.go (hunk 1 above)
Comment (awly) [reply lib/backend/firestore/firestorebk.go]: Good point. In practice it seems to always be UTF-8, but it's not enforced anywhere.
Changed and added test coverage for other backends.

**Pull request merged**